docs/changelog/4_x.rst

```diff
@@ -0,0 +1,454 @@
+
+We've made it easier to work with ranges by adding ``multirange`` as one of
+the datatypes. Multiranges consist of one or more ranges and allow expressing
+intervals that are not contiguous. Multiranges are automatically normalized to
+contain non-overlapping ranges that are ordered according to their boundaries.
+All the usual range functions like ``overlaps`` or ``contains`` work with any
+combination of ranges and multiranges providing more flexibility in
+expressions.
+
+
+Extensions
+==========
+
+auth
+----
+
+The new ``auth`` extension adds a full authentication service that runs
+alongside your database instance saving you the hassle of having to learn and
+implement the intricacies of OAuth or secure password storage.
+
+- OAuth Integration: Seamlessly authenticate with GitHub, Google, Apple, and
+  Azure/Microsoft.
+- Email & Password Support: Includes robust email+password authentication with
+  reset password functionality.
+- Easy Configuration: Set up via our configuration system.
+- Hosted UI: Use our hosted authentication UI to quickly add authentication to
+  your app.
+
+When a user signs up, we create a new object of type ``ext::auth::Identity``,
+which you can link to in your own schema. We then provide you with a token that
+can be set as the global ``ext::auth::client_token`` which will automatically
+populate another computed global called ``ext::auth::ClientTokenIdentity``
+which you can use directly in your access policies, or in your own globals.
+
+.. code-block:: sdl
+
+    using extension auth;
+
+    module default {
+        global current_customer := (
+            assert_single((
+                select Customer { * }
+                filter global ext::auth::ClientTokenIdentity = .identity
+            ))
+        );
+
+        type Customer {
+            required text: str;
+            required identity: ext::auth::Identity;
+        }
+
+        type Item {
+            required sku: str;
+            required description: str;
+        }
+
+        type Cart {
+            required customer: Customer;
+            multi items: Item {
+                quantity: int32;
+            };
+
+            access policy customer_has_full_access
+                allow all
+                using (global current_customer.id ?= .customer.id);
+        }
+    }
+
+
+Here's an example query using the TypeScript client:
+
+.. code-block:: typescript
+
+    import { createClient } from "edgedb";
+
+    declare const tokenFromAuthServer: string;
+    const client = createClient()
+      .withGlobals({
+        "ext::auth::client_token": tokenFromAuthServer
+      });
+
+    const carts = await client.query(`select Cart { * };`);
+
+
+
+pgcrypto
+--------
+
+We've added ``pgcrypto`` to our extensions. This exposes ``digest``, ``hmac``,
+``gen_salt`` and ``crypt`` functions for your hashing, ecrypting and salting
+needs.
+
+.. code-block:: edgeql-repl
+
+  edgedb> select ext::pgcrypto::digest('encrypt this', 'sha1');
```

Comment:
Which hashes are usable?

Reply:
Hi Paul! :-)

> Standard algorithms are md5, sha1, sha224, sha256, sha384 and sha512. Moreover, any digest algorithm OpenSSL supports is automatically picked up.

@vpetrovykh, @raddevon, please make sure this is stated (along with other enumerables at https://www.postgresql.org/docs/current/pgcrypto.html)

Reply:
I've updated the changelog to include this. We can do a separate PR to update the main docs.